src/coreclr/jit/hwintrinsiccodegenarm64.cpp

```diff
@@ -2253,6 +2253,51 @@ void CodeGen::genHWIntrinsic(GenTreeHWIntrinsic* node)
                 break;
             }
 
+            case NI_Sve_ConditionalExtractAfterLastActiveElementScalar:
+            case NI_Sve_ConditionalExtractLastActiveElementScalar:
+            {
+                opt = emitter::optGetSveInsOpt(emitTypeSize(node->GetSimdBaseType()));
+
+                if (emitter::isGeneralRegisterOrZR(targetReg))
```

Comment:
shouldn't this be an assert instead? `assert(emitter::isGeneralRegisterOrZR(targetReg));`?

Reply:
Or are we using the term `*Scalar` for instructions that are generated for https://docsmirror.github.io/A64/2023-06/clasta_r_p_z.html and https://docsmirror.github.io/A64/2023-06/clasta_v_p_z.html?

Reply:
[CLASTA (scalar)](https://docsmirror.github.io/A64/2023-06/clasta_r_p_z.html) is emitted only for integer `*Scalar` APIs. For floating-point `*Scalar` APIs we have to emit [CLASTA (SIMD&FP scalar)](https://docsmirror.github.io/A64/2023-06/clasta_v_p_z.html) instead.